src/session.ts

```diff
@@ -188,6 +199,135 @@ export function valueSession(): void {
+  }
+
+  // Stop tracking meat if we have less than 25 turns left
+  if (
+    get("_garboMarginalMeatCheckpoint") &&
+    get("_garboMarginalMeatTurns") &&
+    continueMeatTracking &&
+    myAdventures() - 25 - globalOptions.saveTurns <= 0
+  ) {
+    continueMeatTracking = false;
+    const { meat, items } = sessionSinceStart().value(garboValue);
+    const meatDiff = meat - get("_garboMarginalMeatCheckpoint", 0);
+    const itemDiff = items - get("_garboMarginalItemCheckpoint", 0);
+    setProperty("_garboMarginalMeatValue", meatDiff.toFixed(0));
+    setProperty("_garboMarginalItemValue", itemDiff.toFixed(0));
+    if (marginalSession) marginalSessionDiff = Session.current().diff(marginalSession);
+  }
+}
+
+function printMarginalSession(): void {
+  if (myInebriety() > inebrietyLimit() || myAdventures() > globalOptions.saveTurns) return;
+
+  if (get("_garboMarginalMeatValue") && get("_garboMarginalMeatTurns")) {
+    const meat = get("_garboMarginalMeatValue", 0);
+    const meatTurns = get("_garboMarginalMeatTurns", 0);
+
+    if (meatTurns <= 0) {
+      print("Error in estimating marginal MPA - meat turns tracked = 0", "red");
+      return;
+    }
+
+    const MPA = meat / meatTurns;
+
+    // Only evaluate item outliers if we have run a good number of turns (to reduce variance)
+    if (
+      marginalSessionDiff &&
+      barfSession &&
+      numTrackedItemTurns &&
+      numTrackedItemTurns >= Math.max(50, meatTurns)
+    ) {
+      const { items, itemDetails } = marginalSessionDiff.value(garboValue);
+      const barfItemDetails = Session.current().diff(barfSession).value(garboValue).itemDetails;
+      const outlierItemDetails = itemDetails
+        .filter(
+          (detail) =>
+            detail.quantity === 1 &&
+            detail.value >= 1000 &&
+            barfItemDetails.some((d) => d.item === detail.item && d.quantity <= 2)
+        )
+        .sort((a, b) => b.value - a.value);
+      print(`Outliers:`, HIGHLIGHT);
+      let outlierItems = 0;
+      for (const detail of outlierItemDetails) {
+        print(
+          `${detail.quantity} ${detail.item} worth ${detail.value.toFixed(0)} total`,
+          HIGHLIGHT
+        );
+        outlierItems += detail.value;
+      }
+      const outlierIPA = items / numTrackedItemTurns;
+      const IPA = (items - outlierItems) / numTrackedItemTurns;
+      const totalOutlierMPA = MPA + outlierIPA;
+      const totalMPA = MPA + IPA;
+      print(
+        `Marginal MPA: ${formatNumber(Math.round(MPA * 100) / 100)} [raw] + ${formatNumber(
+          Math.round(IPA * 100) / 100
+        )} [items] (${formatNumber(
+          Math.round(outlierIPA * 100) / 100
+        )} [outliers]) = ${formatNumber(Math.round(totalMPA * 100) / 100)} [total] (${formatNumber(
+          Math.round(totalOutlierMPA * 100) / 100
+        )} [w/ outliers])`,
+        HIGHLIGHT
+      );
+    } else if (get("_garboMarginalItemValue")) {
+      const items = get("_garboMarginalMeatValue", 0);
```

Comment:
... are we setting this pref? this way to get items/IPA seems very bizarre to me. wouldn't this just straight up double count MPA?

Reply:
Ah this is just a stupid error on my part, it should `get("_garboMarginalItemValue")` instead